torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -273,14 +254,33 @@ void ProcessGroupAgent::sync() {
 }
 
 void ProcessGroupAgent::start() {
+  rpcRunning_.store(true);
```

Comment:
I changed the `futureTimeoutCV_.wait_for(...)` part in https://github.com/pytorch/pytorch/pull/30355.

We need to change this line to
```
std::lock_guar<std::mutex> futureLock{futureMutex_};
rpcRunning_.store(true);
```
in #30355 as well.